api/api/model_endpoints_api.go

```diff
@@ -166,26 +152,25 @@ func (c *ModelEndpointsController) UpdateModelEndpoint(r *http.Request, vars map
 		// Use default environment if not specified
 		env, err = c.EnvironmentService.GetDefaultEnvironment()
 		if err != nil {
-			return InternalServerError("Default environment not found, please specify one")
+			return InternalServerError(fmt.Sprintf("Default environment not found, please specify one: %v", err))
 		}
 		newEndpoint.EnvironmentName = env.Name
 	} else {
 		// Check environment exists
 		env, err = c.AppContext.EnvironmentService.GetEnvironment(newEndpoint.EnvironmentName)
 		if err != nil {
-			if !gorm.IsRecordNotFoundError(err) {
-				log.Errorf("Unable to find the specified environment: %s. Err: %s", newEndpoint.EnvironmentName, err)
-				return InternalServerError(fmt.Sprintf("Unable to find the specified environment: %s", newEndpoint.EnvironmentName))
+			if gorm.IsRecordNotFoundError(err) {
+				return NotFound(fmt.Sprintf("Environment not found: %v", err))
 			}
+			return InternalServerError(fmt.Sprintf("Error getting environment: %v", err))
 
-			return NotFound(fmt.Sprintf("Environment not found: %s", newEndpoint.EnvironmentName))
 		}
 	}
 	newEndpoint.Environment = env
 
 	currentEndpoint, err := c.ModelEndpointsService.FindByID(ctx, modelEndpointID)
 	if err != nil {
-		return NotFound(fmt.Sprintf("Model Endpoint with given `model_endpoint_id: %d` not found", modelEndpointID))
+		return NotFound(fmt.Sprintf("Environment not found: %v", err))
```

Comment:
Should there be a `if gorm.IsRecordNotFoundError(err)` check here, failing which we return `NotFound(fmt.Sprintf("Model endpoint not found: %v", err))`, otherwise we return `InternalServerError(fmt.Sprintf("Error getting model endpoint: %v", err))`? In any case, the error message shouldn't be about an `environment` 😅